ccm_web/server/src/api/api.course.handler.ts

```diff
@@ -0,0 +1,126 @@
+      return CourseApiHandler.slimCourse(response.body)
+    } catch (error) {
+      const errResponse = handleAPIError(error, JSON.stringify(courseData))
+      return { statusCode: errResponse.canvasStatusCode, errors: [errResponse] }
+    }
+  }
+
+  async getSections (): Promise<CanvasCourseSection[] | APIErrorData> {
+    try {
+      const endpoint = `courses/${this.courseId}/sections`
+      const queryParams = { include: ['total_students'] } // use list for "include" values
+      logger.debug(`Sending request to Canvas (get all pages) - Endpoint: ${endpoint}; Method: GET`)
+      const sectionsFull = await this.requestor.list<CanvasCourseSection>(endpoint, queryParams).toArray()
+      logger.debug('Received response (status code unknown)')
+      return sectionsFull.map(s => ({
+        id: s.id,
+        name: s.name,
+        course_id: s.course_id,
+        total_students: s.total_students
+      }))
+    } catch (error) {
+      const errResponse = handleAPIError(error)
+      return { statusCode: errResponse.canvasStatusCode, errors: [errResponse] }
+    }
+  }
+
+  async getSectionsWithCourse (course: CanvasCourseBase): Promise<CourseWithSections | APIErrorData> {
+    const sectionsResult = await this.getSections()
+    if (isAPIErrorData(sectionsResult)) return sectionsResult
+    return { ...course, sections: sectionsResult }
+  }
+
+  async createSection (sectionName: string): Promise<CanvasCourseSection | APIErrorData> {
+    try {
+      const endpoint = `courses/${this.courseId}/sections`
+      const method = HttpMethod.Post
+      const requestBody = { course_section: { name: sectionName } }
+      logger.debug(`Sending request to Canvas - Endpoint: ${endpoint}; Method: ${method}; Body: ${JSON.stringify(requestBody)}`)
+      const response = await this.requestor.requestUrl<CanvasCourseSection>(endpoint, method, requestBody)
+      logger.debug(`Received response with status code ${response.statusCode}`)
+      const newFullSection = response.body
+      return { ...SectionApiHandler.slimSection(newFullSection), total_students: 0 }
+    } catch (error) {
+      const errResponse = handleAPIError(error, sectionName)
+      return { statusCode: errResponse.canvasStatusCode, errors: [errResponse] }
+    }
+  }
+
+  async createSections (sections: string[]): Promise<CanvasCourseSection[] | APIErrorData> {
+    const start = process.hrtime()
+    const apiPromises = sections.map(async (section) => await this.createSection(section))
+    const sectionsOrErrorDataObjs = await Promise.all(apiPromises)
+    // https://codezup.com/measure-execution-time-javascript-node-js/
+    const stop = process.hrtime(start)
+    logger.debug(`Time taken to create ${sections.length} sections: ${(stop[0] * 1e9 + stop[1]) / 1e9} seconds`)
+    return makeResponse<CanvasCourseSection>(sectionsOrErrorDataObjs)
+  }
+
+  async mergeSections (sectionIds: number[]): Promise<CanvasCourseSectionBase[] | APIErrorData > {
```

Comment:
@ssciolla Technically you can cross-list across terms. Like Summer 2021 to Spring 2020 etc. 
I tested the use-case from the PR it allows to cross-list and decross list multiple terms from Swagger-UI

I know our workflow only allows to presents sections from same terms. Should be allow more checks if the both course and section belongs to same term?

But this shouldn't be show stopper to merge this. This just needs more thoughts in the BE and user can't so this from the UI

Reply:
We could limit that, although it would probably involve more API calls to check the course and section.

I guess my thinking at this point is that if the user has modify permissions for the course and section (which scoped permissions will enforce), there's not technically any harm to keeping this possible. They would be able to do this through the API if they minted their own token and hit the Canvas API directly. Besides, I think it would be non-trivial for a user to hit our API directly, they'd have to find the API routes and the CSRF token.

Reply:
OK, We can leave it for now and document it

Reply:
I think we ever open the Swagger API to Prod environment for Functional team, then I think we need to think back on all these things.

Reply:
I'm not sure we'll ever want to do that, but we can discuss.

Reply:
I am not saying opening the swagger UI for this first phase. I don't want to give the access to even dev other than local development. 

But there was some discussion around this early on among dev making it available in Prod for future. So I have to mention this now from that perspective.

Reply:
Gotcha. Was there something else I needed to do here documentation wise?

Reply:
I wanted to document, that BE allows for cross-listing through swagger UI. But the FE flow disallows for user from UI

Reply:
K, not sure where this should be documented, in code or not. I'll leave it as for now.